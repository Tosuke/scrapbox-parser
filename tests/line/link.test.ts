/* global describe it expect */
import '../jest-setup'

describe('link', () => {
  it('Simple absolute link', () => {
    expect('https://example.com/').toMatchSnapshotWhenParsing()
  })

  it('Simple absolute link with bracket', () => {
    expect('[https://example.com/]').toMatchSnapshotWhenParsing()
  })

  it('Simple root link', () => {
    expect('[/project/page]').toMatchSnapshotWhenParsing()
  })

  it('Simple relative link', () => {
    expect('[page]').toMatchSnapshotWhenParsing()
  })

  it('Link with content', () => {
    expect(`[https://example.com/   Example]
[Example   https://example.com/]
[https://left.com/ center https://right.com/]`).toMatchSnapshotWhenParsing()
  })

  it('Root and relative link path can include space', () => {
    expect(`[page name]
[/project/page name]`).toMatchSnapshotWhenParsing()
  })

  it('Link with link', () => {
    expect('[https://example.com https://example.com]').toMatchSnapshotWhenParsing()
  })

  it('Link with GET parameters', () => {
    expect('[http://example.com?key1=value1&key2=value2]').toMatchSnapshotWhenParsing()
  })
})
