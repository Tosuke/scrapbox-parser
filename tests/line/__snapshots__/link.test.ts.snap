// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link Link with GET parameters: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "http://example.com?key1=value1&key2=value2",
        "pathType": "absolute",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Link with content: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "Example",
        "href": "https://example.com/",
        "pathType": "absolute",
        "type": "link",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "Example",
        "href": "https://example.com/",
        "pathType": "absolute",
        "type": "link",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "center https://right.com/",
        "href": "https://left.com/",
        "pathType": "absolute",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Link with link: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "https://example.com",
        "href": "https://example.com",
        "pathType": "absolute",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Root and relative link path can include space: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "page name",
        "pathType": "relative",
        "type": "link",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "/project/page name",
        "pathType": "root",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Simple absolute link with bracket: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "https://example.com/",
        "pathType": "absolute",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Simple absolute link: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "https://example.com/",
        "pathType": "absolute",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Simple relative link: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "page",
        "pathType": "relative",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;

exports[`link Simple root link: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "/project/page",
        "pathType": "root",
        "type": "link",
      },
    ],
    "type": "line",
  },
]
`;
